warp10/src/main/java/io/warp10/script/MemoryWarpScriptStack.java

```diff
@@ -564,165 +563,200 @@ public void exec(String line, long lineNumber) throws WarpScriptException {
     try {
       recurseIn();
 
-      String[] statements;
-
-      line = line.trim();
+      int start = 0;
 
       //
-      // Replace whitespaces in Strings with '%20'
+      // Fast path to process multiline strings or comments blocks
       //
-
-      line = UnsafeString.sanitizeStrings(line);
-
-      if (-1 != line.indexOf(' ') && !inMultiline.get()) {
-        //statements = line.split(" +");
-        statements = UnsafeString.split(line, ' ');
-      } else {
-        // We're either in multiline mode or the line had no whitespace inside
-        statements = new String[1];
-        if (inMultiline.get()) {
-          // If the line only contained the end of multiline indicator with possible wsp on both sides
-          // then set the statement to that, otherwise set it to the raw line
-          if(WarpScriptStack.MULTILINE_END.equals(line)) {
-            statements[0] = line;
+      if (inMultiline.get()) {
+        line = line.trim();
+        // End of multiline        
+        if (WarpScriptStack.MULTILINE_END.equals(line)) {
+          inMultiline.set(false);
+          String mlcontent = multiline.toString();
+          if (null != secureScript) {
+            secureScript.append(" ");
+            secureScript.append("'");
+            try {
+              secureScript.append(WarpURLEncoder.encode(mlcontent, StandardCharsets.UTF_8));
+            } catch (UnsupportedEncodingException uee) {
+            }
+            secureScript.append("'");
           } else {
-            statements[0] = rawline;
+            if (macros.isEmpty()) {
+              this.push(mlcontent);
+            } else {
+              macros.get(0).add(mlcontent);
+            }
           }
+          multiline.setLength(0);
         } else {
-          statements[0] = line;
+          // Append current line to existing multiline 
+          if (multiline.length() > 0) {
+            multiline.append("\n");
+          }
+          multiline.append(rawline);
+        }
+        return;
+      } else if (inComment.get()) {
+        int end = line.indexOf(COMMENT_END);
+        if (-1 == end) {
+          return; // No end of comment in this line, skip it
+        } else {
+          start = end; // Skip the beginning of the line, before */
         }
       }
-
-      //
-      // Report progress
-      //
-
-      progress();
-
+      
       //
-      // Loop over the statements
+      // Process line character by character, looking at block comments, comments, strings, then process statements.
       //
-
-      for (int st = 0; st < statements.length; st++) {
-        handleSignal();
```

Comment:
You removed the call to handleSignal, it must be re-introduced.

Reply:
I moved it, but it is there. 
I check it at the beginning of the for loop (so, between each statement or comment block).

Reply:
my mistake. reintroduced.